server/app/services/geo/esri/EsriServiceAreaValidationConfig.java

```diff
@@ -132,12 +132,41 @@ public Optional<ImmutableMap<String, EsriServiceAreaValidationOption>> getImmuta
    * Creates a list of {@link EsriServiceAreaValidationOption}s with the same URL as the passed in
    * EsriServiceAreaValidationOption, given a map returned from {@link getImmutableMap}.
    */
-  public ImmutableList<EsriServiceAreaValidationOption> mapToListWithSameServiceAreaOptionUrl(
-      EsriServiceAreaValidationOption serviceAreaOption,
-      ImmutableMap<String, EsriServiceAreaValidationOption> immutableMap) {
-    return immutableMap.values().stream()
-        .filter(option -> option.getUrl().equals(serviceAreaOption.getUrl()))
-        .collect(ImmutableList.toImmutableList());
+  public Optional<ImmutableList<EsriServiceAreaValidationOption>> getOptionsWithSharedBackend(
+      String serviceAreaUrl) {
+    if (!isConfigurationValid()) {
+      logger.error(
+          "Error calling EsriServiceAreaValidationConfig.getOptionsWithSharedBackend. Error: Esri"
+              + " Address Service Area Config is missing settings.");
+      return Optional.empty();
+    }
```

Comment:
**suggestion (blocking)** the code that consumes this calls `get()` immediately so will throw a `NullPointerException` if `Optional.empty()` is returned. Recommend throwing an exception here instead and returning `ImmutableList<EsriServiceAreaValidationOption>`.

In general would prefer to avoid using `Optional` for indicating errors in favor of throwing checked exceptions.

Reply:
That's true, though the code above where this is consumed first does a check with `!esriServiceAreaValidationConfig.isConfigurationValid()` and returns empty if not valid, so it's technically guarded. That said, working with all of these optionals is a pain and it would be nicer to throw an exception when `EsriServiceAreaValidationConfig` is initialized and the config is not valid.

Reply:
Added `InvalidEsriServiceAreaValidationConfigException` and updated `getOptionsWithSharedBackend` and `getImmutableMap` to throw if the config is invalid.

Reply:
Ah right on, sorry I missed the guard there. Thanks for the change!